src/main/kotlin/ui/PRComponent.kt

```diff
@@ -147,47 +127,83 @@ class OwnPRComponent(ownPR: PR,
                      awtExecutor: Executor)
     : PRComponent(ownPR, imagesSource, awtExecutor) {
 
-    override fun addApproveButton(buttonSize: Dimension) {
+    override fun createApproveButton() {
         //Do not add an approve button for own PRs
     }
 }
 
-class ReviewersPanel() : JPanel(FlowLayout(FlowLayout.LEFT, 2, 2)) {
-    private val approvers = JPanel(FlowLayout(FlowLayout.LEFT, 2, 2))
-    private val reviewers = JPanel(FlowLayout(FlowLayout.LEFT, 2, 2))
-    private val approvedBy = JBLabel("Approved:")
-    private val reviewBy = JBLabel("Reviewers:")
-    private val others = JBLabel("Others:")
+class ReviewersPanel(reviewers: MutableList<PRParticipant>,
+                     imagesSource: MediaSource<BufferedImage>,
+                     awtExecutor: Executor) : JPanel(HorizontalLayout(5)) {
+    companion object {
+        const val ALWAYS_DISPLAY_REVIEWERS_COUNT = 5
+    }
 
     init {
-        approvedBy.isVisible = false
-        others.isVisible = false
-        reviewBy.isVisible = false
-        add(approvedBy)
-        add(approvers)
-        add(others)
-        add(reviewBy)
-        add(reviewers)
+        this.isOpaque = false
+        reviewers.sortWith(Comparator { o1, o2 -> o1.status.compareTo(o2.status) })
+        val labels: Map<PRParticipant, ReviewerItem> = reviewers.associateWith { prParticipant -> ReviewerItem(prParticipant) }
+
+        val alwaysVisibleReviewerCount = Math.min(ALWAYS_DISPLAY_REVIEWERS_COUNT, reviewers.size)
+
+        reviewers.take(alwaysVisibleReviewerCount).forEach { this.add(labels[it]) }
+
+        val reviewersInCombo = reviewers.size - alwaysVisibleReviewerCount
+        if (reviewersInCombo > 0) {
+            val otherReviewersButton = JLayeredPane()
+            val realButton = JButton("+$reviewersInCombo")
+            val avatarSize = ReviewerComponentFactory.avatarSize
+            realButton.setBounds(0, ReviewerComponentFactory.statusIconSize / 3, avatarSize, avatarSize)
+            otherReviewersButton.add(realButton)
+            this.add(otherReviewersButton)
+            val height = this.preferredSize.height
+            otherReviewersButton.preferredSize = Dimension(height, height)
+            val menu = JBPopupMenu()
+            reviewers.takeLast(reviewersInCombo).forEach { prParticipant: PRParticipant ->
+                val itemPanel = JPanel(FlowLayout(FlowLayout.LEFT))
+                itemPanel.add(labels[prParticipant])
+                itemPanel.add(JLabel(prParticipant.user.displayName))
+                menu.add(itemPanel)
+            }
+            realButton.addMouseListener(object : MouseAdapter() {
+                override fun mouseClicked(e: MouseEvent) {
+                    menu.show(otherReviewersButton, e.x, e.y)
```

Comment:
This is minor too:
If we want to mimic Bitbucket behaviour, we may want to make this button as a toggle, so the second click would hide the pop-up menu.

Reply:
If you do not mind, I would leave it like this until we are working on mimic navigation.

In addition, it is possible to use the “Esc” key to close the pop-up window.